flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/SavepointObserver.java

```diff
@@ -106,36 +117,36 @@ private Optional<String> observeTriggeredSavepointProgress(
             LOG.debug("Savepoint not in progress");
             return Optional.empty();
         }
-        LOG.info("Observing savepoint status");
-        SavepointFetchResult savepointFetchResult;
-        try {
-            savepointFetchResult =
-                    flinkService.fetchSavepointInfo(
-                            currentSavepointInfo.getTriggerId(), jobID, deployedConfig);
-        } catch (Exception e) {
-            LOG.error("Exception while fetching savepoint info", e);
-            return Optional.empty();
-        }
-
-        if (!savepointFetchResult.isTriggered()) {
-            String error = savepointFetchResult.getError();
-            if (error != null
-                    || SavepointUtils.gracePeriodEnded(
-                            configManager.getOperatorConfiguration(), currentSavepointInfo)) {
-                String errorMsg = error != null ? error : "Savepoint status unknown";
-                LOG.error(errorMsg);
+        LOG.info("Observing savepoint status.");
+        SavepointFetchResult savepointFetchResult =
+                flinkService.fetchSavepointInfo(
+                        currentSavepointInfo.getTriggerId(), jobID, deployedConfig);
+
+        if (savepointFetchResult.isPending()) {
+            if (SavepointUtils.gracePeriodEnded(
+                    configManager.getOperatorConfiguration(), currentSavepointInfo)) {
+                String errorMsg =
+                        "Savepoint triggering timed out after "
+                                + configManager
+                                        .getOperatorConfiguration()
+                                        .getSavepointTriggerGracePeriod();
                 currentSavepointInfo.resetTrigger();
+                LOG.error(errorMsg);
                 return Optional.of(errorMsg);
+            } else {
+                LOG.info("Savepoint operation not running yet, waiting within grace period...");
```

Comment:
not running yet -> not finished yet? The savepoint should already be triggered successfully.

Reply:
fixed